### PR TITLE
Add GPUSupportEnabled to ECS configurations

### LINKED ISSUE #2086: [ECS] Conditionally set GPUSupportEnabled in ecs-settings-applier
We need to conditionally set the value of the `GPUSupportEnabled` flag in `ecs-settings-applier` at compilation time, depending on what type of image is being built (accelerated vs general purpose).

This is to support #1074

### PR DESCRIPTION
**Pull request by arnaldo2792**
**Issue number:**
Closes #2086 

**Description of changes:**

In preparation for #1074, this adds the `GPUSupportEnabled` flag as `false` for the existing ECS variant.

```
Now the value of the `GPUSupportEnabled` flag will be set at
compile-time depending on the variant being built.
```

**Testing done:**

I checked that the `ecs.config.json` file was created with the expected flag set as `false` for the general purpose variant ECS variant:

```
[root@admin]# cat /.bottlerocket/rootfs/etc/ecs/ecs.config.json  | jq
{
  "Cluster": "bottlerocket",
  "InstanceAttributes": {
    "bottlerocket.variant": "aws-ecs-1"
  },
  "PrivilegedDisabled": true,
  "AvailableLoggingDrivers": [
    "json-file",
    "awslogs",
    "none"
  ],
  "TaskIAMRoleEnabled": true,
  "TaskIAMRoleEnabledForNetworkHost": true,
  "SELinuxCapable": true,
  "OverrideAWSLogsExecutionRole": true,
  "TaskENIEnabled": true,
  "GPUSupportEnabled": false
}

```

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: sources/api/ecs-settings-applier/src/ecs.rs [hunk 1]
```diff
@@ -101,6 +109,8 @@ async fn run() -> Result<()> {
 
         // awsvpc mode is always available
         task_eni_enabled: true,
+
+        gpu_support_enabled: GPU_SUPPORT_ENABLED,
```
Comment (etungsten) [inline sources/api/ecs-settings-applier/src/ecs.rs]: nit: I wonder if we can make the condition more clear by using the [`cfg!`](https://doc.rust-lang.org/std/macro.cfg.html) macro here:
```suggestion
        gpu_support_enabled: !(cfg!(variant_type = "general_purpose")),
```

then we can ditch the intermediary const bool.

File: sources/api/ecs-settings-applier/src/ecs.rs [hunk 2]
```diff
@@ -64,6 +71,7 @@ pub(crate) async fn main() -> () {
 
 async fn run() -> Result<()> {
     let args = parse_args(env::args());
+    SimpleLogger::init(LevelFilter::Info, LogConfig::default()).context(error::LoggerSnafu)?;
```
Comment (etungsten) [inline sources/api/ecs-settings-applier/src/ecs.rs:71]: I guess logging was never working prior to this change?

nit: Can we move these changes to a separate commit?

Comment (arnaldo2792) [reply sources/api/ecs-settings-applier/src/ecs.rs:71]: That's correct, nothing was logged out. I can put this in a different commit

File: sources/api/ecs-settings-applier/src/ecs.rs (hunk 1 above)
Comment (arnaldo2792) [reply sources/api/ecs-settings-applier/src/ecs.rs]: I can give this a spin :+1:

**Comment by arnaldo2792** [comment]
Last two forced pushes address:

- Remove constant value and use `!cfg` instead
- Separate logging change into its own commit

**Comment by arnaldo2792** [comment]
(Forced push to rebase develop)

File: sources/api/ecs-settings-applier/build.rs [hunk 3]
```diff
@@ -9,6 +9,17 @@ fn main() {
...
+        println!(
+            "cargo:rustc-cfg=variant_family=\"{}\"",
+            parts[0..2].join("-")
+        );
+        let variant_type = if parts.len() > 3 {
+            parts[3]
+        } else {
+            "general_purpose"
+        };
+        println!("cargo:rustc-cfg=variant_type=\"{}\"", variant_type);
```
Comment (webern) [inline sources/api/ecs-settings-applier/build.rs:22]: It doesn't have to be part of this PR, but we should generalize this logic. Possibly buildsys could have a lib.rs that we use for things like this.

**Review by etungsten — APPROVED** [review]

File: sources/api/ecs-settings-applier/build.rs [hunk 4]
```diff
@@ -9,6 +9,17 @@ fn main() {
...
+        let parts = variant.split('-').collect::<Vec<&str>>();
+        println!(
+            "cargo:rustc-cfg=variant_family=\"{}\"",
+            parts[0..2].join("-")
+        );
+        let variant_type = if parts.len() > 3 {
+            parts[3]
+        } else {
+            "general_purpose"
+        };
```
Comment (bcressey) [inline sources/api/ecs-settings-applier/build.rs:21]: Why not check for NVIDIA specifically? Is it because we'd also set `GPUSupportEnabled` for AMD and Inferentia?

**Review by bcressey — APPROVED** [review]

File: sources/api/ecs-settings-applier/build.rs (hunk 4 above)
Comment (arnaldo2792) [reply sources/api/ecs-settings-applier/build.rs:21]: I didn't see support for AMD in the ECS agent, but to enable inferentia support, a different flag has to be (`InferentiaSupportEnabled`) set and we will set it based on the variant's name.

Comment (bcressey) [reply sources/api/ecs-settings-applier/build.rs:21]: It might be better to pin it to NVIDIA if `GPUSupportEnabled` is synonymous with that hardware.

Comment (arnaldo2792) [reply sources/api/ecs-settings-applier/build.rs:21]: But we still need to know what "variant type" is built, since we could have `aws-k8s-*-inferentia` so that in `ecs-settings-applier` we do:

```rust
{
    GPUSupportEnabled : cfg!(variant_type = "nvidia"), 
    InferentiaSupportEnabled: cfg!(variant_type = "inferentia")
}
```

**Comment by arnaldo2792** [comment]
Forced push changes the condition to set `GPUSupportEnabled`

**Review by etungsten — APPROVED** [review]

**Review by bcressey — APPROVED** [review]

**Review by webern — APPROVED** [review]
